### PR TITLE
chore(validation): state rule cleanup

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

### CODE REVIEW

File: packages/validation/src/rules.ts [hunk 1]
```diff
@@ -64,10 +64,14 @@ export const ValidationRuleAliasMessage = Object.freeze({
       };
       aliases = toArray(Object.entries(allMessages)).map(([name, defaultMessage]) => ({ name, defaultMessage }));
     }
-    defineMetadata(aliases, rule instanceof Function ? rule : rule.constructor, this.aliasKey);
+    defineMetadata(aliases, rule, this.aliasKey);
   },
   getDefaultMessages<TRule extends IValidationRule>(rule: Constructable<TRule> | TRule): ValidationRuleAlias[] {
-    return getMetadata(this.aliasKey, rule instanceof Function ? rule : rule.constructor)!;
+    const messages = getMetadata<ValidationRuleAlias[]>(this.aliasKey, rule)!;
```
Comment (bigopon) [inline packages/validation/src/rules.ts:70]: 👍

File: packages/validation/src/rules.ts [hunk 2]
```diff
@@ -246,23 +257,37 @@ export class EqualsRule extends BaseValidationRule implements IEqualsRule {
...
 
-export class StateRule<TValue = any, TObject extends IValidateable = IValidateable, TState = unknown> extends BaseValidationRule<TValue, TObject> {
+export class StateRule<TValue = any, TObject extends IValidateable = IValidateable, TState extends PropertyKey = PropertyKey> extends BaseValidationRule<TValue, TObject> {
   public static readonly $TYPE: string = 'StateRule';
 
+  private _explicitMessageKey: string | null = null;
+
+  protected _messageKey!: string;
+  public get messageKey(): string { return this._explicitMessageKey ?? this._messageKey ?? (void 0)!; }
+  public set messageKey(value: string) { this._explicitMessageKey = value; }
```
Comment (bigopon) [inline packages/validation/src/rules.ts:267]: shouldn't this belong to the base class? It doesn't seem invalid there.

**Review by bigopon — APPROVED** [review]
Nice work @Sayan751

File: packages/validation/src/rules.ts (hunk 2 above)
Comment (Sayan751) [reply packages/validation/src/rules.ts:267]: The base rule already has the infra. No change is required there. Made some minor clean up.

**Review by bigopon — APPROVED** [review]

**Pull request merged**